signingscript/src/signingscript/sign.py

```diff
@@ -811,27 +812,49 @@ def write_signing_req_to_disk(fp, signing_req):
     Does proper base64 and json encoding.
     Tries not to hold onto a lot of memory.
     """
```

Comment:
As mentioned in channel: I'm not 100% sure what we're doing in `write_signing_req_to_disk` and why. I believe this affects all autograph signing formats, and when we roll out we'll affect all trains/products/releases, so we probably a) want to do more testing than just debsign testing before rolling this out, and b) choose a non-release non-nightly time and do some doublechecking before the next nightly/release.

This may be worth pushing to `dev-signingscript` if we haven't already (I'm guessing you have already :)
When we do, we can also try some other test signing tasks, e.g. point some gecko Windows or other signing tasks at the dev pool.

Also, if we're able to add some unit tests to this function before and after the refactor, that may help us reason about the changes here a bit better. (I'm thinking we have some example signing reqs, and some expected file contents to diff against.)

Reply:
I am not sure why it builds the byte string this way. So I didn't change it. I refactored the code (renamed `write_signing_req_to_disk` to `_encode_single_file`) so we can see the logic hasn't changed in the diff. The new code I had was getting 400s from autograph... I reverted the logic to be equivalent to the original commit (build the byte string in a similar way as `_encode_single_file`). Autograph is happy now.

I think it is a good idea to do some testing on non deb sign tasks & rolling it out at a quiet time. `dev-signingscript` is in sync with this branch ([here's a task completed by the latest version of the code](https://firefox-ci-tc.services.mozilla.com/tasks/G3njVLRVRvGjmMw6R1lgOQ))

I will add some unit tests. From the coverage report it seems most of the uncovered code it's the new code I added.